src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -176,43 +176,32 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
                     {
                         serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
                     }
-                }
-                catch
-                {
-                }
 
-                try
-                {
                     if ((string.IsNullOrEmpty(serializerPath) || !File.Exists(serializerPath)) && !string.IsNullOrEmpty(Assembly.GetEntryAssembly().Location))
                     {
                         serializerPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), serializerName + ".dll");
                     }
+
+                    if (!string.IsNullOrEmpty(serializerPath))
+                    {
+                        serializer = Assembly.LoadFile(serializerPath);
+                    }
                 }
-                catch
-                {
-                }
- 
-                if (!string.IsNullOrEmpty(serializerPath))
+                catch (Exception e)
                 {
-                    try
+                    if (e is OutOfMemoryException)
```

Comment:
Should we check ThreadAbortException and StackOverfowException as well like what we did in full framework?
http://referencesource.microsoft.com/#System.Xml/System/Xml/Serialization/Compilation.cs,176